Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -722,8 +713,8 @@ - (void)didChange:(NSKeyValueChange)changeKind valuesAtIndexes:(NSIndexSet*)inde
 - (void)willChangeValueForKey:(NSString*)key withSetMutation:(NSKeyValueSetMutationKind)mutationKind usingObjects:(NSSet*)objects {
     if ([self observationInfo]) {
         NSKeyValueChange changeKind = _changeFromSetMutationKind(mutationKind);
-        NSMutableDictionary* change = [NSMutableDictionary dictionaryWithObject:@(changeKind) forKey:NSKeyValueChangeKindKey];
-        _dispatchWillChange(self, key, [mutationKind, objects, change, changeKind](_NSKVOKeyObserver* keyObserver) {
+        _dispatchWillChange(self, key, [mutationKind, objects, changeKind](_NSKVOKeyObserver* keyObserver) {
```

Comment:
it will be worth testing this to see what happens for the `key.collection.key` case -- I think you're supposed to get a `set` notification since `.key` is a calculated subcollection.